.github/workflows/ci_cd.yml

```
@@ -67,6 +85,8 @@ jobs:
           cache-from: type=gha,scope=${{ matrix.image }}
           cache-to: type=gha,scope=${{ matrix.image }}
           outputs: type=docker,dest=/tmp/${{ matrix.image }}.tar
+          build-args: |
+            SEMANTIC_VERSION=${{ needs.get-image-tag.outputs.image-tag }}
```

Shouldn't this start with `steps`?
```suggestion
            SEMANTIC_VERSION=${{ steps.get-image-tag.outputs.image-tag }}
```

---

Nevermind! I found the difference:
- `needs` is for different `jobs`
- `steps` is for different steps inside the same job

Reference: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs

---

That's exactly it!